public/app/features/explore/utils/logsVolumeResponse.ts

```diff
@@ -0,0 +1,19 @@
+import { DataQueryError, DataQueryResponse } from '@grafana/data';
+
+// Currently we can only infer if an error response is a timeout or not.
+// TODO: add backend support for specifying the type of error.
+export function isTimeoutErrorResponse(response: DataQueryResponse | undefined): boolean {
+  if (!response) {
+    return false;
+  }
+  if (!response?.error && !response?.errors) {
+    return false;
+  }
+
+  const errors = response?.error ? [response?.error] : response?.errors || [];
+
+  return errors.reduce((isTimeoutError: boolean, error: DataQueryError) => {
```

Comment:
I am wondering if this wouldn't be "faster"/better with a [`some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some)?

Reply:
Definitely, good idea.